lib_eio/unix/process.ml

```diff
@@ -69,23 +69,49 @@ let get_env = function
   | Some e -> e
   | None -> Unix.environment ()
 
-class virtual mgr = object (self)
-  inherit Eio.Process.mgr
+type mgr_ty = [`Mgr_unix | Eio.Process.mgr_ty]
```

Comment:
We might want to copy the ``[`Platform of 'tag | ...]`` pattern used in net. That allows saying that a Unix process manager spawns Unix processes.